light-client/tests/integration.rs

```diff
@@ -91,16 +86,21 @@ fn sync() {
         clock_drift: Duration::from_secs(5 * 60),      // 5 minutes
     };
 
-    let primary_instance = make_instance(primary, options, io.clone());
-    let witness_instance = make_instance(witness, options, io);
+    let rpc_client = rpc::HttpClient::new(node_address).unwrap();
+    let primary_io = ProdIo::new(primary, rpc_client.clone(), Some(Duration::from_secs(2)));
+    let witness_io = ProdIo::new(witness, rpc_client, Some(Duration::from_secs(2)));
+
+    let primary_instance = make_instance(primary, options, primary_io);
+    let witness_instance = make_instance(witness, options, witness_io);
 
-    let peer_list = PeerList::builder()
-        .primary(primary, primary_instance)
-        .witness(witness, witness_instance)
-        .build();
+    let peer_list = {
+        let mut builder = PeerList::builder();
+        builder.primary(primary, primary_instance);
+        builder.witness(witness, witness_instance);
+        builder.build()
```

Comment:
Can't these still be chained?

Reply:
Not since the API change in 26d1efe4c3ce5bbc6436c0deb8f2e83bbfb00ef9.

I figured I would follow the [current recommendations](https://docs.rs/derive_builder/0.9.0/derive_builder/#mutable-aka-non-consuming-recommended) of the popular `derive_builder` crate. I agree that the chained version looks a lot better, so I would definitely not mind going back on that.